projects/media-viewer/src/lib/store/store-utils.ts

```diff
@@ -26,6 +27,52 @@ export class StoreUtils {
       }, {});
   }
 
+  static genTagNameEntities(annotations) {
+    const filterAnnoWithNoCommentsTags = annotations.filter(a => (a.comments.length && a.tags.length));
+    const allTags = filterAnnoWithNoCommentsTags.map(anno => this.groupByKeyEntities(anno.tags, 'name'));
+    const groupedByName = allTags.reduce(
+      (tagEntitiy: { [id: string]: Annotation }, tagItem) => {
+        return {
+          ...tagEntitiy,
+          ...tagItem
+        };
+      }, {});
+
+    return this.genNameEnt(annotations, groupedByName)
+  };
+
+  static genNameEnt(annos, groupedByName) {
+     return Object.keys(groupedByName).reduce(
+      (tagNameEnt, key) => {
+        const annot = annos.map(anno => {
+          const tags = anno.tags.map(tag => {
+            if (tag.name === key) {
+              return anno.id;
+            }
+          });
+          return tags.filter(a => a !== undefined);
+        });
+        const readyAnno = annot.reduce((obj: {[id: string]: string}, anno: string[]) => {
+          const a = anno.length ? anno[0] : null;
+          if (a) {
+            return {
+              ...obj,
+              [a]: a
+            };
+          } else {
+            return {
+              ...obj
+            };
+          }
+        }, {})
```

Comment:
so is readyAnno an object where the key is the annoId and the value is the annoId

Reply:
i think if u think this can be done better we need to pair up... is much more complex then that... but there might be a better solution... i am not very happy but it works and it gives me what I want - yeah is the brain/logic  of the whole filtering

Reply:
this seems to work:
return Object.keys(groupedByName).reduce(
      (tagNameEnt, key) => {
        const readyAnno = annos.filter(anno => anno.tags.find(tag => tag.name === key))
          .map(anno => anno.id)
          .reduce((obj: {[id: string]: string}, anno: string) => ({ ...obj, [anno]: anno }), {});
        return {
          ...tagNameEnt,
          [key]: readyAnno
        }
      }, {});